### ISSUE TITLE
Add a tag for version used in osqp python bindings

### ISSUE THREAD

**Issue by h-vetinari** [osqp/osqp#497]
Currently, the last released version of osqp-python [uses](https://github.com/osqp/osqp-python/blob/v0.7.9/CMakeLists.txt#L7-L10) a branch from this repository (in addition to including it as a git submodule at a _different_ commit).

This makes it very hard to disentangle the various layers of qdldl, osqp & python bindings, which we have to do in conda-forge to package these libraries.

It would be very helpful to have a tag for https://github.com/osqp/osqp/commit/e7bd8c6442b72d9b224007c424718e1bca6fea32 that has _some_ version number - whether that's 0.7.9, or 1.0.0.alpha0, or ...

Then we'd have something to package and use for the python-bindings.

Medium term, it would of course be amazing to see osqp 1.0 come to fruition, but I don't want to nag about the schedule for that - just tagging what's used in the last release of osqp-python would be good enough.

**Comment by h-vetinari** [comment]
Ping @vineetbansal @bstellato @gbanjac @SteveDiamond @rileyjmurray; the osqp in conda-forge is not compatible with scipy 1.10 anymore, and trying to [fix](https://github.com/conda-forge/osqp-feedstock/pull/48) this runs into this issue. Could someone please help here? 

Details are in the OP, but if nothing else, then at least:

> It would be very helpful to have a tag for https://github.com/osqp/osqp/commit/e7bd8c6442b72d9b224007c424718e1bca6fea32 that has _some_ version number - whether that's 0.7.9, or 1.0.0.alpha0, or ...

**Comment by vineetbansal** [comment]
@h-vetinari. Thanks for the ping, and sorry for the delay here. I can pin `osqp-python` to a specific commit hash of `osqp` (which will be on the `develop-1.0` branch) and do a pre-release of version 1.0.0 this weekend (say `1.0.0.alpha1`). Would that solve your problem?

**Comment by h-vetinari** [comment]
> Would that solve your problem?

Yes, as long as the commit used in osqp-python is tagged in osqp, we can deal with it. Thanks!

**Comment by h-vetinari** [comment]
> I can pin `osqp-python` to a specific commit hash of `osqp` (which will be on the `develop-1.0` branch) and do a pre-release of version 1.0.0 this weekend (say `1.0.0a1`). Would that solve your problem?

Hey @vineetbansal, AFAICT no new tag has been pushed yet. It would be nice to finally unblock newer osqp-python (& Python 3.11 support) in conda-forge.

**Comment by h-vetinari** [comment]
@vineetbansal, after I saw that 0.6.3 was released just now, I went to check the status of osqp-python, which still [uses](https://github.com/osqp/osqp-python/blob/v0.7.9/CMakeLists.txt#L6-L10) the develop branch in its last available release. I also saw that the `CMakeLists.txt` in question has been deleted on master. Is there any timeline for a now osqp-python release which uses a released osqp under the hood.

**Comment by vineetbansal** [comment]
Hi @h-vetinari - I plan on making a `0.6.3` release of `osqp-python`, which will point to the `0.6.3` tag of `osqp` that was released today (as a git submodule as is done for all `0.x` releases). This will be the same process as the existing `0.6.2.post9` release, which points to the `0.6.2` tag of `osqp`.

Over the course of the next couple of days a `1.0.0.beta0` version of `osqp` will be released, and a corresponding `1.0.0.beta0` of `osqp-python` which will point to that tag in its `CMakeLists.txt` (no more git submodules).

So what I'm trying to understand here is - how are you calling `v0.7.9` as the **last** available release? The whole `v0.7.x` series of tags were for internal use only while we figured out our CI workflow, never published to pyPI, and certainly never intended to be published on `conda-forge`. Maybe we threw you off with the numbering? What if I told you that `v0.6.2.post9` *is* the latest release as of right now? Will that help in fixing your workflow?

In any case, both versions `0.6.3` (which will be the last `0.x` version) and version `1.0.0.beta0` of both `osqp` and `osqp-python` should be available by this weekend.

**Comment by h-vetinari** [comment]
Great to hear!



> So what I'm trying to understand here is - how are you calling `v0.7.9` as the **last** available release? The whole `v0.7.x` series of tags were for internal use only while we figured out our CI workflow, never published to pyPI, and certainly never intended to be published on `conda-forge`.

Good to know. For context, a published tag on github that's not somehow marked as a prerelease, alpha or beta, _is_ a release. Some projects only do their releases like that. So apologies for not knowing that idiosyncrasy of your development model.

You're most welcome to join the feedstock maintainer team, so you'd be in the loop about what we do with osqp in conda-forge. I [invited](https://github.com/osqp/osqp-python/issues/32) the osqp developers at the time and Bartolomeo and Goran took me up on that - though they don't seem that active anymore here either.

**Comment by vineetbansal** [comment]
Thanks @h-vetinari . Good to know - I'll be more careful with these "experimental" tags going forward.

I've just released `v0.6.3` to PyPI. Let me know if I should _also_ tag it as `v0.7.10` so your process can pick it up / you can whitelist it again. We don't anticipate using `0.x` versions anymore (and in the remote chance that we do, will skip `0.7.x` entirely).

The upcoming `1.0.0.beta0` will be marked as beta after all, so it will not show up on conda-forge (if my understanding is correct) anyway.

**Comment by h-vetinari** [comment]
> I've just released `v0.6.3` to PyPI.

Cool!

> Let me know if I should _also_ tag it as `v0.7.10` so your process can pick it up

The way I understood it, we don't want to publish that anyway? I'll wait for 1.0.0. 

> Over the course of the next couple of days a `1.0.0.beta0` version of `osqp` will be released, and a corresponding `1.0.0.beta0` of `osqp-python` which will point to that tag in its `CMakeLists.txt` (no more git submodules).

While it's definitely better to have only one (not both submodule and `FetchContent` in Cmake), I think a submodule is much more explicit and reproducible. I'll be able to deal with it in either case, but from the outside (not knowing what benefits you expect for your development) this appears worse than the current situation.
